challenge-7.js

```diff
@@ -0,0 +1,193 @@
+const warehouse = Warehouse([]);
+
+function Product(id, name, price, count, sell = [], enable = true) {
+  if (count === 0) enable = false;
+  return {
+    id,
+    name,
+    count,
+    price,
+    enable,
+    sell,
+  };
+}
+
+function Warehouse(products = []) {
+  return products;
+}
+
+function warehouseStock() {
+  let productsCount = 0;
+  for (const product of warehouse) {
+    productsCount += product.count;
+  }
+  return productsCount;
+}
+
+function productStock(productId) {
+  if (!isTypeValid(productId, "number")) return "id should be a number!";
+  for (const product of warehouse) {
+    if (product.id === productId) {
+      return product;
+    }
+  }
+  return `product ${productId} is not available!`;
+}
+
+function addNewProduct(product) {
+  if (findProduct(product.id)) return "id should be unique!";
+  warehouse.push(product);
+  return product.name + " is added!";
+}
+
+function sellProduct(productId, count = 1) {
+  if (!isTypeValid(productId, "number")) return "id should be a number!";
```

Comment:
Better to write another function to evaluate the entry data and use it here instead of having three if statements.